src/algebra/geom_sum.lean

```diff
@@ -229,6 +229,28 @@ begin
     sub_sub_sub_cancel_right (x ^ n) (x ^ m * y ^ (n - m)) (y ^ n)],
 end
 
+protected theorem commute.geom_sum₂_add_mul_eq_geom_sum₂ {α : Type u} [ring α] {x y : α}
+  (h : commute x y) {n : ℕ} :
+  x ^ n + y * (geom_series₂ x y n) = (geom_series₂ x y (n + 1)) :=
```

Comment:
I think this lemma would be much easier to name in the other direction, as
```suggestion
protected theorem commute.geom_series₂_succ_eq {α : Type u} [ring α] {x y : α}
  (h : commute x y) {n : ℕ} :
  geom_series₂ x y (n + 1) = x ^ n + y * geom_series₂ x y n :=
```

Reply:
That's a bit of the tail wagging the dog, but I like it.